src/user.rs

```diff
@@ -13,144 +15,233 @@ use async_trait::async_trait;
+/// Options for device creation.
+///
+/// Default values are provided with [DeviceCreateOpts::default()](#method.default)
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct DeviceCreateOpts {
     device_name: Option<DeviceName>,
 }
 impl DeviceCreateOpts {
-    /// Create a new device with an optional readable name for the device.
+    /// # Arguments
+    /// - device_name
+    ///   - `None` (default) - The device will be created with no name.
+    ///   - `Some` - The provided name will be used as the device's name.
     pub fn new(device_name: Option<DeviceName>) -> DeviceCreateOpts {
         DeviceCreateOpts { device_name }
     }
 }
 impl Default for DeviceCreateOpts {
+    /// Default `DeviceCreateOpts` for common use cases.
+    ///
+    /// The device will be created with no name.
     fn default() -> Self {
         DeviceCreateOpts::new(None)
     }
 }
 
-/// Options that can be specified at when calling [`user_create`](trait.UserOps.html#tymethod.user_create).
+/// Options for user creation.
+///
+/// Default values are provided with [UserCreateOpts::default()](#method.default)
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct UserCreateOpts {
-    // see docs on `new`
     needs_rotation: bool,
 }
 
 impl UserCreateOpts {
-    /// Constructor. Also see [`default`](#method.default).
-    ///
     /// # Arguments
-    /// - `needs_rotation` - Set to true if the private key for this user should be rotated when the user "takes control" of their keys.
-    ///
-    /// The main use case for this is a workflow that requires that users and groups to be generated
-    /// prior to the user logging in for the first time. In this situation, a user's cryptographic identity
-    /// can be generated by a third party, like a server process, and then the user can take control of their
-    /// keys by rotating the private key using [`rotate_private_key`](https://github.com/IronCoreLabs/ironoxide/issues/44).
+    /// - `needs_rotation` - `true` if the private key for this user marked for rotation
     pub fn new(needs_rotation: bool) -> UserCreateOpts {
         UserCreateOpts { needs_rotation }
     }
 }
 
 impl Default for UserCreateOpts {
+    /// Default `UserCreateOpts` for common use cases.
+    ///
+    /// The user will be created with their private key not marked for rotation.
     fn default() -> Self {
         UserCreateOpts::new(false)
     }
 }
 
 /// IronOxide User Operations
+///
+/// # Key Terms
+/// - Device - The only entity in the Data Control Platform that can decrypt data. A device is authorized using a user’s private key,
+///     therefore a device is tightly bound to a user.
+/// - ID - The ID representing a user or device. It must be unique within its segment and will **not** be encrypted.
+/// - Password - The string used to encrypt and escrow a user's private key.
+/// - Rotation - Changing a user's private key while leaving their public key unchanged. The main use case for this is a workflow
```

Comment:
I think the usecase portion might be more at home on `user_rotate_private_key` and `group_rotate_private_key` with links to those at the top of the user/group modules, perhaps under the rotation sections...

Reply:
I did look on the `group` module and there's no mention of rotation there in the Key Terms. I guess we should add it if it's going to be in the `user` module.

Reply:
I think that the `Rotation` section should stay (and I should add one to the `group` module as well) because `needsRotation` is so prevalent in creation types and return types. But I agree that the use case part can move to the functions themselves. 
I'll make that change in my next commit